### DISCUSSION TITLE
How to close dialog and sheet

### DISCUSSION THREAD

**Discussion by koalacapek** [spartan-ng/spartan#520] [Q&A] [answered]
How do we close sheet and dialog on button click

**Comment by Georg632** [comment, marked as answer]
### BrnDialogCloseDirective

Add `BrnDialogCloseDirective` to your imports and add the selector to your button:
```
      <button
        hlmBtn
        brnDialogClose
        type="submit">
        Save changes
      </button>
```

### By Code

Add template reference or use viewchild to gain access to the dialog:
`<hlm-dialog #dialog>`
Use `close` function:
```
      <button
        hlmBtn
        (click)="dialog.close({})"
        type="submit">
        Save changes
      </button>
```

Sheet is basically the same, just different names.

**Reply by goetzrobin** [reply]
@Georg632 would you be down to add this to the documentation page in a PR?

**Reply by Georg632** [reply]
@goetzrobin Yeah, I would be down. Tried to create a branch with the changes but I don't have the permission to do that.
Sorry, first time contributing to something public.

**Reply by goetzrobin** [reply]
No worries! And welcome to becoming an open source contributor! You'll have to fork the repository, which means you'll get your own copy of the repository on your GitHub account. Then you'll make and push your changes to that repository and open a PR to spartan! 

Couple helpful links:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo

https://gist.github.com/jagregory/710671

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-repository-for-a-fork
